### PR TITLE
Let Super User view and edit segments created by other users

### LINKED ISSUE #6766: Let Super User view and edit segments created by other users
A Super admin is not able to list all segments available on the Piwik instance via UI.
Since Segments can have a rather large impact on the archiving / performance of a Piwik instance, it would be useful to see what segments are available on the system.

### PR DESCRIPTION
**Pull request by mattab**
As a Super User:
- I can now see all segments that were created for this website by any other user
- When a segment was created by another user who is not Super User, the segment appears below a new section "Visible to you because you have Super User access:"
- Such segments are editable by the Super User
- The only difference when editing someone else's segment, as a Super User, is that "This segment is visible to [ME]" now says "This segment is visible to [SEGMENT_AUTHOR_USERNAME]"
- One can now search in the  search bar for a username and see all segments created by this user

For all users:
- New section "Shared with you:" now lists segments created by a Super User, and marked as "Visible to [All Users]"
- Before segments  shared with me, looked the same as segments I created, which was confusing

Fixes #6766

### CODE REVIEW

**Comment by sgiehl** [comment]
Maybe unrelated to this issue. But imho we should adjust the hover effect when choosing a segment or a website. Both should look the same. Currently a hovered website has a grey background and a hovered segment has a grey border only. Shall we create a new issue for that? Or simply adjust it within this one?

File: plugins/SegmentEditor/javascripts/Segmentation.js [hunk 1]
```diff
@@ -80,14 +80,14 @@ Segmentation = (function($) {
...
-                    var name = $(foundItems).first().find("span.segname").text();
-                    title.text(name);
+                    var idSegment = $(foundItems).first().attr('data-idsegment');
+                    var title = getSegmentName( getSegmentFromId(idSegment));
                 } else {
-                    title.text(_pk_translate('SegmentEditor_CustomSegment'));
+                    title = _pk_translate('SegmentEditor_CustomSegment');
                 }
-                segmentationTitle.html(title);
+                segmentationTitle.html( "<strong>" + title + "</strong>");
```
Comment (sgiehl) [inline plugins/SegmentEditor/javascripts/Segmentation.js]: Wouldn't it be better to remove the `<strong>` here and to add a css class (having `font-weight: bold`) instead. That would make it easier for custom designs to overwrite the css at that point.

Note: The `.text()` below does not remove a `<strong>` (if set before), it simply replaces the text within. Not sure if that is expected behavior.

**Comment by sgiehl** [comment]
Besides the comment, LGTM :shipit:

**Comment by mattab** [comment]
> Maybe unrelated to this issue. But imho we should adjust the hover effect when choosing a segment or a website. Both should look the same. Currently a hovered website has a grey background and a hovered segment has a grey border only. Shall we create a new issue for that? Or simply adjust it within this one?

@sgiehl good point, I never noticed this! could you create a new issue or PR for it?

**Pull request merged**